src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -610,146 +610,160 @@ public string[] Name
+            }
+
+            try
+            {
+                foreach (ServiceController service in MatchingServices())
                 {
-                    if (DependentServices.IsPresent)
+                    if (!DependentServices.IsPresent && !RequiredServices.IsPresent)
                     {
-                        foreach (ServiceController dependantserv in service.DependentServices)
+                        WriteObject(AddProperties(scManagerHandle, service));
+                    }
+                    else
+                    {
+                        if (DependentServices.IsPresent)
                         {
-                            WriteObject(dependantserv);
+                            foreach (ServiceController dependantserv in service.DependentServices)
+                            {
+                                WriteObject(dependantserv);
+                            }
                         }
-                    }
 
-                    if (RequiredServices.IsPresent)
-                    {
-                        foreach (ServiceController servicedependedon in service.ServicesDependedOn)
+                        if (RequiredServices.IsPresent)
                         {
-                            WriteObject(servicedependedon);
+                            foreach (ServiceController servicedependedon in service.ServicesDependedOn)
+                            {
+                                WriteObject(servicedependedon);
+                            }
                         }
                     }
                 }
             }
+            finally
+            {
+                if (scManagerHandle != nint.Zero)
+                {
+                    bool succeeded = NativeMethods.CloseServiceHandle(scManagerHandle);
+                    Diagnostics.Assert(succeeded, "SCManager handle close failed");
+                }
+            }
         }
 
         #endregion Overrides
 
+#nullable enable
         /// <summary>
         /// Adds UserName, Description, BinaryPathName, DelayedAutoStart and StartupType to a ServiceController object.
         /// </summary>
+        /// <param name="scManagerHandle">Handle to the local SCManager instance.</param>
         /// <param name="service"></param>
         /// <returns>ServiceController as PSObject with UserName, Description and StartupType added.</returns>
-        private PSObject AddProperties(ServiceController service)
+        private static PSObject AddProperties(nint scManagerHandle, ServiceController service)
         {
-            NakedWin32Handle hScManager = IntPtr.Zero;
             NakedWin32Handle hService = IntPtr.Zero;
-            int lastError = 0;
-            PSObject serviceAsPSObj = PSObject.AsPSObject(service);
+
+            // As these are optional values, a failure due to permissions or
+            // other problem is ignored and the properties are set to null.
+            bool isDelayedAutoStart = false;
```

Comment:
Does it make sense to have the default value for `isDelayedAutoStart` to be `null` too? In case we failed to retrieve the actual value for this property, we probably should not use `false` as the default value.

Reply:
I can make it nullable, it just means that we would need to decide how `GetServiceStartupType` treats the value as `null`. We could either

+ Treated as false
+ Treated the whole start type as invalid (or when `Automatic`)

The former is simpler but it may not be true if the service is truly automatic but delayed and we couldn't get the correct result.

https://github.com/PowerShell/PowerShell/blob/27e8cb3c9212d1d3e9a14198cbe822e7f7aec9dc/src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs#L2915-L2931

Reply:
Has been updated to keep it as `InvalidValue` (what it was if the config couldn't be retrieved) if we couldn't get the automatic delayed start value.